service/history/workflow/task_refresher.go

```diff
@@ -162,7 +154,7 @@ func (r *TaskRefresherImpl) RefreshTasks(
 	)
 }
 
-func (r *TaskRefresherImpl) refreshTasksForWorkflowStart(
```

Comment:
It still makes sense for this function to be part of the TaskRefresher interface. If this function doesn't need to be a part of TaskRefresher, the same applies to RefreshTasks() also.

Reply:
can you explain why it should be part of TaskRefresher interface?

Reply:
I don't see any reason for that. The only reason I see is, it should be grouped with RefreshTasks().
In that case, it looks like it is better to get rid of TaskRefresher altogether.

Reply:
Here is a long explanation.
RefreshTasksForWorkflowStart needs taskGenerator.  Mutable state has taskGenerator.

But TaskRefresherImpl:
1. Create new taskGenerator (will it be the same? probably, but who knows)
2. Need shardContext, that is not needed for RefreshTasksForWorkflowStart

So options are :

1. 
   a. add RefreshTasksForWorkflowStart to TaskRefresher interface, then introduce subtle side-effect (it is using different generator. may work for now, but this is still side-effect, and just clumsy). 
   b. find a way to "pass through" shard context and logger to that call, even if both those parameters are not used in RefreshTasksForWorkflowStart.

2. Don't do that, just define RefreshTasksForWorkflowStart function that will accept only needed arguments.

Option 1. is very clumsy. So I choose 2.